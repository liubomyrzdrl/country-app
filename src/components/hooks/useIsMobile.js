import { useState, useEffect } from "react";

export const useIsMobile = () => {
  const [windowWidth, setWindowWidth] = useState();
  const WIDTH_DEVICE = 996;

  useEffect(() => {
    const handleResize = () => {
      setWindowWidth(window.innerWidth);
    }

    window.addEventListener("resize", handleResize);
    handleResize();

    return () => window.removeEventListener("resize", handleResize);
  }, []);

  return windowWidth <= WIDTH_DEVICE;
};
