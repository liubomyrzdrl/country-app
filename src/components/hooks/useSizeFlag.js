import { useState, useEffect } from "react";

export const useSizeFlag = () => {
  const [sizeFlag, setSizeFlag] = useState();
  const size = {
    LARGE: "large",
    MEDIUM: "medium",
    SMALL: "small",
  };

  useEffect(() => {
    const handleResize = () => {
      if (window.innerWidth > 475) {
        setSizeFlag(size.LARGE);
      }

      if (window.innerWidth <= 475 && window.innerWidth > 375) {
        setSizeFlag(size.MEDIUM);
      }

      if (window.innerWidth <= 375) {
        setSizeFlag(size.SMALL);
      }
    };

    window.addEventListener("resize", handleResize);
    handleResize();

    return () => window.removeEventListener("resize", handleResize);
  }, [size.LARGE, size.MEDIUM, size.SMALL]);

  return sizeFlag;
};
