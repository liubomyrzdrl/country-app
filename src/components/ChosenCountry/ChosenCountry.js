import React from 'react';
import { gql } from '@apollo/client';
import PropTypes from 'prop-types';
import { WithQuery } from '../hoc/withQuery';
import { ChosenCountryDesktop } from './ChosenCountryDesktop/ChosenCountryDesktop';
import { ChosenCountryMobile } from './ChosenCountryMobile/ChosenCountryMobile';
import { useIsMobile } from '../hooks/useIsMobile';

const COUNTRY__QUERY = gql`
  query getCountry($code: ID!) {
    country(code: $code) {
      code
      name
      native
      phone
      continent {
        name
      }
      capital
      currency
      languages {
        name
      }
    }
  }
`;

export const ChosenCountry = WithQuery(({ data }) => {
  const isMobile = useIsMobile();
  return !isMobile ? (
    <ChosenCountryDesktop data={data} />
  ) : (
    <ChosenCountryMobile data={data} />
  );
}, COUNTRY__QUERY);

ChosenCountry.propTypes = {
  data: PropTypes.object,
};
