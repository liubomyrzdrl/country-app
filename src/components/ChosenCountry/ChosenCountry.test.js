import React from 'react';
import { render } from '@testing-library/react';
import { MockedProvider } from '@apollo/client/testing';
import { useIsMobile } from '../hooks/useIsMobile';
import { ChosenCountry } from './ChosenCountry';

const mocks = [];
jest.mock('../hooks/useIsMobile.js', () => ({
  useIsMobile: jest.fn(),
}));

describe('Test ChosenCountry component', () => {
  test('it should render desktop version ChosenCountry component', () => {
    useIsMobile.mockImplementation(() => false);
    const { container } = render(
      <MockedProvider mocks={mocks} addTypename={false}>
        <ChosenCountry />
      </MockedProvider>
    );
    expect(container).toMatchSnapshot();
  });

  test('it should render mobile version ChosenCountry component', () => {
    useIsMobile.mockImplementation(() => true);
    const { container } = render(
      <MockedProvider mocks={mocks} addTypename={false}>
        <ChosenCountry />
      </MockedProvider>
    );
    expect(container).toMatchSnapshot();
  });
});
