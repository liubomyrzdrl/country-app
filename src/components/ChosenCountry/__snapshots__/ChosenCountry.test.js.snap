// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test ChosenCountry component it should render desktop version ChosenCountry component 1`] = `
<div>
  <div
    class="spinner"
  />
</div>
`;

exports[`Test ChosenCountry component it should render mobile version ChosenCountry component 1`] = `
<div>
  <div
    class="spinner"
  />
</div>
`;

exports[`Test ChosenCountry component it should render mobile version ChosenCountryDesktop component 1`] = `
<div>
  <div
    class="spinner"
  />
</div>
`;
